src/main/java/net/minecraftforge/common/ForgeVersion.java

```diff
@@ -204,6 +207,46 @@ public void run()
                 }
             }
 
+            /**
+             * Opens stream for given URL while following redirects (max. 5)
+             */
+            private InputStream openUrlStream(URL url) throws IOException
+            {
+                URL originalUrl = url;
+                int redirects = 0;
+                do
+                {
+                    URLConnection c = url.openConnection();
+                    boolean redirected = false;
+
+                    if (c instanceof HttpURLConnection)
+                    {
+                        HttpURLConnection huc = (HttpURLConnection) c;
+                        int responseCode = huc.getResponseCode();
+                        if (responseCode >= 301 && responseCode <= 303 || responseCode == 307 || responseCode == 308)
+                        {
+                            String loc = huc.getHeaderField("Location");
+                            url = new URL(loc);
```

Comment:
Use `url = new URL(url, loc)` to support relative URLs.

Reply:
Looking at https://stackoverflow.com/a/26046079, does the header field need URL-decoding too?

Reply:
No. Doing so is stupid. The header field contains a valid URL. URL-encoding is for when you want to safely embed an arbitrary string in an URL. This is not the case here, HTTP header fields are not URLs.

Reply:
The redirect support in HttpURLConnection does not use URL decoding, so it's safe to assume this should not either.